### PR TITLE
Add new scalyr.k8s.enableExplorer chart config option which enables Kubernetes Explorer

### PR DESCRIPTION
**Pull request by Kami**
This pull request adds new ``scalyr.k8s.enableExplorer`` chart config option. When this config option is enabled / set to true, Kubernetes Explorer functionality is enabled in the agent.

NOTE: At this point to get a complete Kubernetes Explorer experience, user still needs to install and configure kubernetes state metrics exporter and node exporter pods to be scraped as per docs (https://app.scalyr.com/help/scalyr-agent-k8s-explorer).

### CODE REVIEW

**Comment by Kami** [comment]
As part of this change I also added some other chart config options which make troubleshooting easier (``scalyr.debugLevel``, ``scalyr.ingestDebugLog`` - those also helped troubleshoot me some issues related to those changes) and I also updated default kubernetes_monitor config to match what's set upstream in the default Kubernetes Docker images (specifically, ``stop_agent_on_failure`` monitor config option being set to ``true``).

File: charts/scalyr-agent/templates/configmap.yaml [hunk 1]
```diff
@@ -17,8 +17,24 @@ data:
         {{- if .Values.scalyr.k8s.enableLogs }}
         {
           "module": "scalyr_agent.builtin_monitors.kubernetes_monitor",
+          "stop_agent_on_failure": true,
```
Comment (Kami) [inline charts/scalyr-agent/templates/configmap.yaml:20]: @ArthurKamalov Just FYI - yet another place we need to keep in sync 🤷

**Comment by Kami** [comment]
I also pushed changes needed to test with the latest Kubernetes version (v1.24.x). I will also go ahead and remove testing with v1.16.x to reduce the size of the build matrix. This way the build matrix will also be consistent with the scalyr agent one.

File: charts/scalyr-agent/templates/configmap.yaml [hunk 2]
```diff
@@ -17,8 +17,25 @@ data:
...
           "report_k8s_metrics": {{ .Values.scalyr.k8s.enableMetrics }}
+        },
+        {{- end }}
+        {{- if .Values.scalyr.k8s.enableExplorer }}
+        {
+          module:  "scalyr_agent.builtin_monitors.kubernetes_openmetrics_monitor",
+          include_node_name: true,
+          include_cluster_name: true,
+          logger_include_node_name: false,
+          // Switch to true once preview phase is over
```
Comment (ArthurKamalov) [inline charts/scalyr-agent/templates/configmap.yaml:31]: Since we've done a stable release, is it time to do that?

**Review by ArthurKamalov — APPROVED** [review]
LGTM

File: charts/scalyr-agent/templates/configmap.yaml (hunk 2 above)
Comment (Kami) [reply charts/scalyr-agent/templates/configmap.yaml:31]: Functionality is still in preview, but we can decide how to handle it in a future release - for production deployments setting it to true is the correct thing to do, but for minikube setups when people are testing it, they would need to set this to false.

We should definitively set it to true when it goes out of the preview phase and add docs on how to disable it for local testing, minikube and similar.

Comment (ArthurKamalov) [reply charts/scalyr-agent/templates/configmap.yaml:31]: Ah, got it. I just connected a new stable release with the end of the preview.

Comment (Kami) [reply charts/scalyr-agent/templates/configmap.yaml:31]: Yep, but I'm 100% on board that the default value should be true once it goes out of the preview :)

**Pull request merged**